elytron/src/main/java/org/wildfly/extension/elytron/SSLDefinitions.java

```diff
@@ -1511,6 +1532,56 @@ protected void installedForResource(ServiceController<SSLContext> serviceControl
         return createSSLContextDefinition(ElytronDescriptionConstants.CLIENT_SSL_CONTEXT, false, add, attributes, serverOrHostController);
     }
 
+    private static final AttributeDefinition[] DYNAMIC_SSL_CONTEXT_ATTRIBUTES = new AttributeDefinition[]{AUTHENTICATION_CONTEXT_ATTRIBUTE};
+
+    static ResourceDefinition getDynamicClientSSLContextDefinition(boolean serverOrHostController) {
+        return createSSLContextDefinition(ElytronDescriptionConstants.DYNAMIC_CLIENT_SSL_CONTEXT, false, new DynamicSSLContextAddHandler(), DYNAMIC_SSL_CONTEXT_ATTRIBUTES, serverOrHostController, Stability.COMMUNITY);
+    }
+
+    private static class DynamicSSLContextAddHandler extends BaseAddHandler {
+        private DynamicSSLContextAddHandler() {
+            super(SSL_CONTEXT_RUNTIME_CAPABILITY, DYNAMIC_SSL_CONTEXT_ATTRIBUTES);
+        }
+
+        @Override
+        protected void performRuntime(OperationContext context, ModelNode operation, ModelNode model) throws OperationFailedException {
+            ServiceBuilder<SSLContext> serviceBuilder = installService(context, model);
+            ServiceName authenticationContextServiceName =
+                    AUTHENTICATION_CONTEXT_RUNTIME_CAPABILITY.getCapabilityServiceName(AUTHENTICATION_CONTEXT_ATTRIBUTE.resolveModelAttribute(context, model).asString());
+            serviceBuilder.requires(authenticationContextServiceName);
+            commonRequirements(serviceBuilder).setInitialMode(ServiceController.Mode.ACTIVE).install();
```

Comment:
Why must this service start ACTIVEly?

Reply:
@pferraro True, I have changed this to be `ON_DEMAND` instead